sos/collector/clusters/ocp.py

```diff
@@ -78,7 +78,7 @@ def oc_cmd(self):
             self._oc_cmd = 'oc'
             if self.primary.host.in_container():
                 _oc_path = self.primary.run_command(
-                    'which oc', chroot=self.primary.host.sysroot
+                    'which oc', need_root=True
```

Comment:
These are very much not the same thing, and this is not the correct fix.

`chroot` means "run this command in a chroot, so that we aren't in `'/`'". The conditional before this makes it obvious that we're trying to make sure that we run in the host's mounted filesystem, not inside the container.

`need_root` simply makes sure we're executing as the root user, or otherwise we use `sudo.

Reply:
They are not the same thing, but run_command() doesn't provide a 'chroot' argument. It does provide a 'need_root' argument, and it's used in other places in ocp.py, like in line #349. Should we instead rewrite run_command() to accept a chroot argument?

Reply:
The correct way is to look in the mounted root filesystem inside the container, which would be passed by the `HOST` or similar env var and be known to sos. This is what the `chroot` is intended for.

The current change does not actually fix that binary discovery. It simply runs `which oc`. The `needs_root` here is meaningless as we'd already be root in the container.

Reply:
Ok, so I'll remove that part and resend the PR with only the fix that solves the backtrace.